var/spack/repos/builtin/packages/nvidia-hpc-sdk/package.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+from spack.util.prefix import Prefix
+import os
+import platform
+
+_versions = {
+    '20.7': {
+        'Linux_aarch64': ('5b83ca1919199ac0aa609309b31c345c5a6453dd3131fddeef9e3ee9059a0e9b', 'https://developer.download.nvidia.com/hpc-sdk/nvhpc_2020_207_Linux_aarch64_cuda_11.0.tar.gz'),
+        'Linux_ppc64le': ('800ead240bdf61611910b2f6df24ee1d7359377ff3767c923738dd81fcea9312', 'https://developer.download.nvidia.com/hpc-sdk/nvhpc_2020_207_Linux_ppc64le_cuda_multi.tar.gz'),
+        'Linux_x86_64':  ('a5c5c8726d2210f2310a852c6d6e03c9ef8c75e3643e9c94e24909f5e9c2ea7a', 'https://developer.download.nvidia.com/hpc-sdk/nvhpc_2020_207_Linux_x86_64_cuda_multi.tar.gz')
+    }
+}
+
+
+class NvidiaHpcSdk(Package):
+    """The NVIDIA HPC SDK is a comprehensive suite of compilers and libraries
+    enabling HPC developers to program the entire HPC platform from the GPU
+    foundation to the CPU and through the interconnect. It is the only
+    comprehensive, integrated SDK for programming accelerated computing
+    systems."""
+
+    homepage = "https://developer.nvidia.com/hpc-sdk"
+
+    maintainers = ['Nischay-Pro']
+
+    for ver, packages in _versions.items():
+        key = "{0}_{1}".format(platform.system(), platform.machine())
```

Comment:
So… no cross-compiling :)

Reply:
yes, no cross compiling but you just have to by the node with target arch :D